pkgs/by-name/me/merge-ut-dictionaries/package.nix

```
@@ -0,0 +1,96 @@
+{
+  lib,
+  fetchFromGitHub,
+  stdenvNoCC,
+  nix-update-script,
+  python3,
+  jawiki-all-titles-in-ns0,
+  ibus-engines,
+
+  mozcdic-ut-jawiki,
+  mozcdic-ut-personal-names,
+  mozcdic-ut-place-names,
+  mozcdic-ut-sudachidict,
+
+  dictionaries ? [
+    mozcdic-ut-jawiki
+    mozcdic-ut-personal-names
+    mozcdic-ut-place-names
+    mozcdic-ut-sudachidict
+  ],
+}:
+
+stdenvNoCC.mkDerivation {
+  pname = "merge-ut-dictionaries";
+  version = "0-unstable-2024-08-28";
+
+  src = fetchFromGitHub {
+    owner = "utuhiro78";
+    repo = "merge-ut-dictionaries";
+    rev = "9cacaadfc4a199914726687ad6545bfd7152c001";
+    hash = "sha256-AdcwXl+33pyN8Q95EDNwMps3ObCzys8nicege6yeRk8=";
+  };
+
+  nativeBuildInputs = [ python3 ];
+
+  preConfigure =
+    if (dictionaries == [ ]) then
+      throw "merge-ut-dictionaries needs at least one dictionary"
+    else
+      ''
```

Not a blocker: to a code reader it might be nice to be told the conditions for the arguments as soon as possible, for code locality and so on. How about this?

```suggestion

assert lib.assertMsg (dictionaries != [ ]) "merge-ut-dictionaries needs at least one dictionary.";

stdenvNoCC.mkDerivation {
  pname = "merge-ut-dictionaries";
  version = "0-unstable-2024-08-28";

  src = fetchFromGitHub {
    owner = "utuhiro78";
    repo = "merge-ut-dictionaries";
    rev = "9cacaadfc4a199914726687ad6545bfd7152c001";
    hash = "sha256-AdcwXl+33pyN8Q95EDNwMps3ObCzys8nicege6yeRk8=";
  };

  nativeBuildInputs = [ python3 ];

  preConfigure = ''
```